File: 2/5.py
def main(n, arr):
    result = [1] * n

    for i in range(n):
        for j in range(n):
            if i != j and arr[i] < arr[j]:
                result[i] += 1

    return result


print(main(5, [87, 89, 92, 100, 76]))
print(main(3, [92, 92, 76]))

File: 2/5.js
// 시간 복잡도: O(n^2)
// 공간 복잡도: O(n)
function main(n, arr) {
  const result = Array.from({ length: n }, () => 1);

  for (let i = 0; i < n; i++) {
    for (let j = 0; j < n; j++) {
      if (i !== j && arr[i] < arr[j]) {
        result[i]++;
      }
    }
  }

  return result;
}

console.log(main(5, [87, 89, 92, 100, 76]));
console.log(main(3, [92, 92, 76]));
